connect_ext_test.go

```diff
@@ -475,7 +476,7 @@ func TestConcurrentStreams(t *testing.T) {
 	server := memhttptest.NewServer(t, mux)
 	var done, start sync.WaitGroup
 	start.Add(1)
-	for i := 0; i < 100; i++ {
+	for i := 0; i < runtime.GOMAXPROCS(0)*2; i++ {
```

Comment:
What's the value in increasing the concurrency further? *2 ensures theres some level of parallelism. If `GOMAXPROCS` is 1, 2 will still be testing parallel execution.

Reply:
I was hoping for stronger guarantee of concurrent activity -- the scheduler is non-deterministic, so it seems trivial for this to allow one goroutine to complete before the other actually starts. I suppose another way to force concurrency might be to call `Send(nil)` (to make sure streams are actually started) before calling `start.Wait()`.

But I guess it also depends on what this is trying to test. The previous value of 100 is quite high, so I assumed it wanted to be a sort of stress test. And in that case, reducing all the way from 100 down to 2 streams seems rather insufficient. So, based on my interpretation of the intent, I would expect to see more than just 2 here.

Reply:
The windows runner has more cores. They are relatively low and maybe would catch more with more testing. Each stream does do [100 ](https://github.com/connectrpc/connect-go/blob/8292c67eaa6150cb524d67e26c9d5406c31b813a/connect_ext_test.go#L486) back on forth on the stream per go rountine. So I thought it would be good enough to just have some parallelism for the test. Will bump to 8 and can always lower later.

Windows:
```
    connect_ext_test.go:514: NUMCPU 4
    connect_ext_test.go:515: GOMAXPROCS 4
```
Linux:
```
    connect_ext_test.go:514: NUMCPU 2
    connect_ext_test.go:515: GOMAXPROCS 2
```